### ISSUE TITLE
[Quantization + FSDP] Support `quantize_()` for DTensor

### ISSUE THREAD

**Issue by gau-nernst** [pytorch/ao#803]
While trying out INT8 mixed precision pretraining (#748) with torchtitan, I came across an issue that if the model is FSDP-sharded, `quantize_()` won't work. The fix would be adding an extra logic to handle DTensor, similar to what FP8 is doing

https://github.com/pytorch/ao/blob/f5703b07acc683653556d04ef970709ba47dba10/torchao/float8/float8_tensor.py#L161-L183

**Comment by msaroufim** [comment]
Yeah this came up in some discussions with inference providers like SGLang as well

**Comment by supriyar** [comment]
@jerryzh168 @kwen2501 is this addressed now with quantize + distributed inference composability work?

**Comment by jerryzh168** [comment]
this is not addressed yet, this is training use case I think, that we can explore in 2025 H1 together with @vkuzo, we do need a guide on how DTensor composes with quantization in both inference and training use cases

**Comment by Andy0422** [comment]
@jerryzh168 is there any update for this issue? Cheers！

**Comment by Andy0422** [comment]
> see some examples in https://github.com/pytorch/ao/blob/main/test/float8/test_fsdp.py
> 
> we'll be using `quantize_` API everywhere, but maybe not yet for
> 
> [ao/test/float8/test_fsdp.py](https://github.com/pytorch/ao/blob/137b0795acb3282ce622948b1537e20914186eea/test/float8/test_fsdp.py#L88)
> 
> Line 88 in [137b079](/pytorch/ao/commit/137b0795acb3282ce622948b1537e20914186eea)
> 
>  convert_to_float8_training( 
> , cc [@vkuzo](https://github.com/vkuzo) [@danielvegamyhre](https://github.com/danielvegamyhre) on the plan to move to quantize_ API there
> what is your use case [@Andy0422](https://github.com/Andy0422)

My use case is inference. I'd like to use quantize_API, such as Int8DynamicActivationInt8WeightConfig, to quantize the weight first, then move the weight and scale with FSDP, hopefully use int8 allreduce.
